.github/workflows/publish-charm.yaml

```diff
@@ -24,6 +24,7 @@ on:
 
 jobs:
   publish-charm:
+    permissions: write-all
```

Comment:
Do we need to give it all the write permissions? I think this could be narrowed down a bit, no?

Reply:
I am not actually sure which specific (lack of) permission is causing the issue, as the logs do not provide anything meaningful other than "Resource not available"

Reply:
Looking at the code for the [action](https://github.com/canonical/charming-actions/blob/main/src/actions/upload-charm/upload-charm.ts), I think the issue is happening when it is at the step of tagging a release on the project. This makes me think that `contents: write` should be enough.